ctapipe/image/extractor.py

```diff
@@ -104,16 +106,69 @@ def neighbor_average_waveform(waveforms, neighbors, lwt):
     return sum_ / n
 
 
-class ChargeExtractor(Component):
+def extract_pulse_time_weighted_average(waveforms):
+    """
+    Use the weighted average of the waveforms to extract the time of the pulse
+    in each pixel
+
+    Parameters
+    ----------
+    waveforms : ndarray
+        Waveforms stored in a numpy array.
+        Shape: (n_chan, n_pix, n_samples)
+
+    Returns
+    -------
+    pulse_time : ndarray
+        Floating point pulse time in each pixel
+        Shape: (n_chan, n_pix)
+
+    """
+    samples_i = np.indices(waveforms.shape)[2]
+    pulse_time = np.average(samples_i, weights=waveforms, axis=2)
```

Comment:
Performance note: line 127 can be pulled out of this function and be done only once per telescope. 
I found it interesting to see, that for waveform shapes like (2, 1000, 300) lines 127 and 128 both took equally long, ~1.7ms or so on my machine. So moving line 127 out of this, should reduce runtime by ~50%.

I found an ugly hack to make `sample_i` in a faster way using something called `stride_tricks` ... I would not recommend using it ... but it might be interesting to know about.
```
In [65]: %%timeit
    ...: i = np.indices(waveforms.shape)[2]
    ...: result = np.average(i, weights=waveforms, axis=2)
3.55 ms ± 16.8 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [63]: %%timeit 
    ...: S = waveforms.shape
    ...: I = np.arange(S[2])
    ...: i = np.lib.stride_tricks.as_strided(I, (S[0], S[1], I.size), (0, 0, I.itemsize))  # a 3D view, no copy of the data
    ...: result = np.average(i, weights=waveforms, axis=2)
1.53 ms ± 5.35 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Reply:
Yep this is a good point. I also noticed that np.indices is surprisingly slow in my investigations for #1038. It would be my intention to replace this function with a numba for-loop version in a future PR, which should result in a huge improvement in performance.

I did not know about `stride_tricks`, thank-you for bringing it up.